lib/azure/armrest/model/base_model.rb

```diff
@@ -33,10 +33,21 @@ class BaseModel < Delegator
       #
       def initialize(json)
         @json = json
+        @resource_group = nil
         @ostruct = JSON.parse(json, object_class: OpenStruct)
         __setobj__(@ostruct)
       end
 
+      # Return the resource group for the current object.
+      def resource_group
+        @resource_group ||= id[/resourceGroups\/(.+?)\//i, 1] rescue nil
+      end
+
+      # Return a hash of tags associated with the resource.
+      def tags
```

Comment:
I think a resource can have anywhere from 10 to 100 tags associated with it, depending on the resource. I don't know how common it will be in the wild, and I think it mostly serves as a form of metadata.